src/go/api/experiment/experiment.go

```diff
@@ -507,22 +506,26 @@ func Start(ctx context.Context, opts ...StartOption) error {
 
 				return errors
 			}
+		}
 
-			if err := app.ApplyApps(ctx, exp, app.Stage(app.ACTIONPOSTSTART), app.DryRun(o.dryrun)); err != nil {
-				errors := multierror.Append(nil, fmt.Errorf("applying apps to experiment: %w", err))
-
-				if err := app.ApplyApps(context.TODO(), exp, app.Stage(app.ACTIONCLEANUP), app.DryRun(o.dryrun)); err != nil {
-					errors = multierror.Append(errors, fmt.Errorf("cleaning up app experiments: %w", err))
-				}
+		if err := app.ApplyApps(ctx, exp, app.Stage(app.ACTIONPOSTSTART), app.DryRun(o.dryrun)); err != nil {
+			errors := multierror.Append(nil, fmt.Errorf("applying apps to experiment: %w", err))
 
-				if err := mm.ClearNamespace(exp.Spec.ExperimentName()); err != nil {
-					errors = multierror.Append(errors, fmt.Errorf("killing experiment VMs: %w", err))
-				}
+			if err := app.ApplyApps(context.TODO(), exp, app.Stage(app.ACTIONCLEANUP), app.DryRun(o.dryrun)); err != nil {
+				errors = multierror.Append(errors, fmt.Errorf("cleaning up app experiments: %w", err))
+			}
 
-				return errors
+			if err := mm.ClearNamespace(exp.Spec.ExperimentName()); err != nil {
```

Comment:
Why are we clearing the namespace if it's a dry run?

Reply:
This line get executed regardless of whether or not this is a dry run. Clearing a namespace that doesn't exist shouldn't cause any errors.